src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateArray = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use a destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, .......... } = action;
    
    switch (type) {
```